pkgs/crypto/crypto.go

```diff
@@ -587,3 +595,92 @@ func EncryptPrivateKey(priv []byte, keyStorePassword string) ([]byte, error) {
 	}
 	return json.Marshal(encryptedData)
 }
+
+func GetPubCommitsFromSharesData(reshare *wire.Reshare) ([]kyber.Point, error) {
+	suite := kyber_bls12381.NewBLS12381Suite()
+	signatureOffset := phase0.SignatureLength
+	pubKeysOffset := phase0.PublicKeyLength*len(reshare.OldOperators) + signatureOffset
+	sharesExpectedLength := EncryptedKeyLength*len(reshare.OldOperators) + pubKeysOffset
+	if len(reshare.Keyshares) != sharesExpectedLength {
+		return nil, fmt.Errorf("GetPubCommitsFromSharesData: shares data len is not correct, expected %d, actual %d", sharesExpectedLength, len(reshare.Keyshares))
+	}
+	pubKeys := utils.SplitBytes(reshare.Keyshares[signatureOffset:pubKeysOffset], phase0.PublicKeyLength)
+	// try to recover commits
+	var kyberPubShares []*share.PubShare
+	for i, pubk := range pubKeys {
+		blsPub := &bls.PublicKey{}
+		err := blsPub.Deserialize(pubk)
+		if err != nil {
+			return nil, err
+		}
+		v := suite.G1().Point()
+		err = v.UnmarshalBinary(blsPub.Serialize())
+		if err != nil {
+			return nil, err
+		}
+		kyberPubhare := &share.PubShare{
+			I: int(i),
+			V: v,
+		}
+		kyberPubShares = append(kyberPubShares, kyberPubhare)
+	}
+	pubPoly, err := share.RecoverPubPoly(suite.G1(), kyberPubShares, int(reshare.OldT), len(reshare.OldOperators))
+	if err != nil {
+		return nil, err
+	}
+	_, commits := pubPoly.Info()
+	return commits, nil
+}
+
+func GetSecretShareFromSharesData(keyshares, initiatorPublicKey, ceremonySigs []byte, oldOperators []*wire.Operator, opPrivateKey *rsa.PrivateKey, operatorID uint64) (*share.PriShare, error) {
+	suite := kyber_bls12381.NewBLS12381Suite()
+	signatureOffset := phase0.SignatureLength
+	pubKeysOffset := phase0.PublicKeyLength*len(oldOperators) + signatureOffset
+	sharesExpectedLength := EncryptedKeyLength*len(oldOperators) + pubKeysOffset
+	if len(keyshares) != sharesExpectedLength {
+		return nil, fmt.Errorf("GetSecretShareFromSharesData: shares data len is not correct, expected %d, actual %d", sharesExpectedLength, len(keyshares))
+	}
```

Comment:
we need it only to inform users that something wrong with the  `keyshares` slice , security wise if something is wrong with keyshares, we will not be able to decrypt it anywhay, and even if one of `oldOperators` cant decrypt the whole protocol will fail and return error. Anyway Ill add more checks here to have it well informed.

Reply:
@pavelkrolevets oh i see, but yes "well-informed" is what i mean — its good to have descriptive errors so users can know they provided wrong order for example instead of 'invalid signature' which is hard to debug

Reply:
@moshe-blox added additional checks for public key position, length etc.